### PR TITLE
Add Hex board

### PR DESCRIPTION
**Pull request by Dilorfin**

### CODE REVIEW

File: sources/hex-board/Board.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+static sf::Vector2i mapAxialToArray(const sf::Vector2i axial)
+{
+	return { axial.x + (int)std::floor(axial.y / 2.f), axial.y };
+}
+static sf::Vector2i mapScreenToArray(const sf::Vector2i mousePosition, const float hexRadius)
+{
+	return mapAxialToArray(mapScreenToAxial(mousePosition, hexRadius));
+}
+
+#include "../resource-manager/ResourceManager.hpp"
```
Comment (toor1245) [inline sources/hex-board/Board.cpp:64]: why is ResourceManager here?

File: sources/hex-board/Board.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	const sf::Vector2f test = sf::Vector2f(mousePosition) + sf::Vector2f(0.f, hexRadius / 2.f);
+
+	sf::Vector2f result;
+	result.x = (test.x * std::sqrt(3.f) - test.y) / 3.f;
+	result.y = 2.f * test.y / 3.f;
+	result /= hexRadius;
+	result = cube_to_axial(cube_round(axial_to_cube(result)));
+
+	return { static_cast<int>(result.x), static_cast<int>(result.y - 1) };
+}
```
Comment (Dilorfin) [inline sources/hex-board/Board.cpp:53]: https://www.redblobgames.com/grids/hexagons/#pixel-to-hex

File: sources/hex-board/Board.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+static sf::Vector2i mapAxialToArray(const sf::Vector2i axial)
+{
+	return { axial.x + (int)std::floor(axial.y / 2.f), axial.y };
+}
```
Comment (Dilorfin) [inline sources/hex-board/Board.cpp:58]: https://www.redblobgames.com/grids/hexagons/#map-storage

File: sources/scenes/AdventureScene.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#include "AdventureScene.hpp"
+
+AdventureScene::AdventureScene(sf::RenderWindow& window)
+	: Scene(window),
+	  board({ 33, 20 })
```
Comment (Dilorfin) [inline sources/scenes/AdventureScene.cpp:5]: `{ 33, 20 }` is just for test purposes

File: sources/scenes/MainMenuScene.cpp [hunk 5]
```diff
@@ -13,6 +15,10 @@ void MainMenuScene::update(const float deltaTime)
 
 void MainMenuScene::handleEvent(const sf::Event& event)
 {
+	if (event.type == sf::Event::KeyPressed && event.key.code == sf::Keyboard::A)
+	{
+		SceneManager::getInstance()->openScene("adventure");
+	}
```
Comment (Dilorfin) [inline sources/scenes/MainMenuScene.cpp:21]: Just to be able to reach adventure scene

File: sources/hex-board/Board.cpp [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	sf::Text text(ResourceManager::getInstance()->getFont());
+	for (size_t y = 0; y < boardSize.y; y++)
+	{
+		for (size_t x = 0; x < boardSize.x; x++)
+		{
+			sf::Color color;
+			color.r = rand() % 255;
+			color.b = rand() % 255;
+			color.g = rand() % 255;
+			hex.setFillColor(color);
```
Comment (Dilorfin) [inline sources/hex-board/Board.cpp:90]: This will be replaced with data got from "array"

File: sources/hex-board/Board.cpp (hunk 1 above)
Comment (Dilorfin) [reply sources/hex-board/Board.cpp:64]: because xD

File: sources/hex-board/Board.cpp [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+			hex.setPosition({ xStep * x + (y % 2) * xStep / 2.f - 5.f, yStep * y });
+
+			text.setPosition(hex.getPosition() + sf::Vector2f(15, 15));
+
+			const sf::Vector2i textPosition(hex.getPosition() + hex.getLocalBounds().getSize() / 2.f);
+			const sf::Vector2i map = mapScreenToArray(textPosition, hexRadius);
+
+			text.setString(std::format("{} {}", map.x, map.y));
```
Comment (Dilorfin) [inline sources/hex-board/Board.cpp:98]: text don't have be here in final version, so it's just for development and will be replaced in further mrs

**Review by toor1245 — APPROVED** [review]

**Pull request merged**